common/cache/domainCache.go

```diff
@@ -80,15 +80,17 @@ type (
 	// it is guaranteed that  CallbackFn pair will be both called or non will be called
 	CallbackFn func(updatedDomains []*DomainCacheEntry)
 
+	CatchUpFn func(domainCache DomainCache, prepareCallback PrepareCallbackFn, callback CallbackFn)
```

Comment:
I would suggest just adding a small one-liner about the intent of this function and it's difference from the prepareCallbck fn